acrg/countrymask.py

```diff
@@ -505,10 +505,40 @@ def create_country_mask(domain,lat=None,lon=None,reset_index=True,ocean_label=Tr
     
     return ds
 
+def mask_fill_gaps(mask_array):
+    
+    #TODO: create a test for fill_gaps
+    print('\nFilling in gaps between masks\nby checking for grid cells where surrounding cells are indentical.')
+    print('WARNING: This is experimental, when using this mask to estimate country emissions from countries'+
+          'with complex coastlines you should manually check the mask before use.')
+    print('WARNING: This does not work for any gaps that are on the edge of the domain.')
+        
+    # loop through grid to check for missing cells
+
+    for i in np.arange(1,mask_array.shape[0]-1):
+        for j in np.arange(1,mask_array.shape[1]-1):
+
+            if mask_array[i,j] == 0.:
+                    
+                # find all surrounding grid cells
+                surrounding = np.delete(mask_array[i-1:i+2,j-1:j+2].flatten(),4)
+
+                # if all surrounding cells are identical, fill in the gap
+                if np.all(surrounding == surrounding[0]) and surrounding[0] != 0.:
+
+                    mask_array[i,j] = surrounding[0]
+
+                # if all but one of the surrounding cells are identical, fill in the gap
+                # (fills in gaps of two adjacent grid cells)
+                elif np.count_nonzero(surrounding == stats.mode(surrounding)[0][0]) == 7. and stats.mode(surrounding)[0] != 0.:
+
+                    mask_array[i,j] = stats.mode(surrounding)[0][0]
+                    
+    return mask_array
 
 def create_country_mask_eez(domain,include_land_territories=True,
```

Comment:
For this function, I think it would be useful to allow explicit "lat" and "lot" arrays to be included as well as just the domain. Not urgent, but this would also match against the options for the other `create_country_mask` function if that was the aim.

Reply:
I've updated this function to include an option to create a countrymask over any lat/lon array.